src/Clash/Cores/ClashFloPoCo/GenTemDSL.hs

```diff
@@ -0,0 +1,1306 @@
+module Clash.Cores.ClashFloPoCo.GenTemDSL
+  ( getPipeDep,
+    flopocoPrim,
+    genFloPoCoInfoEntity,
+    liftInfoEntity,
+    genBlackBoxFunction,
+    genTemplateFunction,
+    genBlackBoxTemplateFunction,
+    genBlackBoxTemplateFunctionProd,
+    genBlackBox,
+    genBlackBoxProd,
+  )
+where
+
+import Clash.Annotations.Primitive (Primitive (..))
+import Clash.Backend (Backend)
+import qualified Clash.Cores.ClashFloPoCo.FloPoCoCall as FPCC
+import Clash.Cores.ClashFloPoCo.InfoEn
+import Clash.Cores.ClashFloPoCo.Lexer
+import Clash.Netlist.BlackBox.Types
+  ( BlackBoxFunction,
+    BlackBoxMeta (..),
+    TemplateKind (..),
+    emptyBlackBoxMeta,
+  )
+import qualified Clash.Netlist.Id as Id
+import Clash.Netlist.Types
+  ( BlackBox (..),
+    BlackBoxContext,
+    EntityOrComponent (..),
+    TemplateFunction (..),
+  )
+import qualified Clash.Netlist.Types as NT
+import qualified Clash.Primitives.DSL as DSL
+import Control.Monad.State (liftIO)
+import Control.Monad.State.Lazy (State)
+import Data.Char (toLower)
+import Data.Maybe (fromMaybe)
+import Data.String.Interpolate (__i)
+import Data.Text (Text)
+import Data.Text.Prettyprint.Doc.Extra (Doc)
+import GHC.Stack (HasCallStack)
+import Language.Haskell.TH.Syntax
+import Text.Show.Pretty (ppShow)
+import Prelude
+
+-- | Function to convert pipeline depth from InfoEntity to Nat number
+getPipeDep :: InfoEntity -> Q Type
+getPipeDep infoen = do
+  return (LitT (NumTyLit (fromIntegral (convertMaybeToInt (pipedep infoen)))))
+
+-- | Function to generate the InfoEntity coresponding to the arithmetic entity generated by FloPoCo
+--
+--  __Example:__
+--
+--  > floPoCoPath = "/home/user/flopoco/build/bin/flopoco"
+--  > args = ["frequency=100", "target=Zynq7000", "IEEEFPAdd", "wE=8", "wF=23","name=plusFloat", "outputFile=flopocoAdd.vhdl","registerLargeTables=1"]
+--  > filePath = "flopocoAdd.vhdl"
+--  > infoEnplus = $(genFloPoCoInfoEntity floPoCoPath args filePath)
+genFloPoCoInfoEntity ::
+  -- | path where FloPoCo is built or installed
+  String ->
+  -- | input to feed FloPoCo
+  [String] ->
+  -- | Path to the VHDL file generated by FloPoCo
+  String ->
+  Q Exp
+genFloPoCoInfoEntity floPoCoPathName argsName fileName = do
+  output <- liftIO $ FPCC.callFloPoCoWithInput floPoCoPathName argsName ""
+  if "Error" `elem` lines output
+    then fail "Error in FloPoCo output"
+    else do
+      result <- processVHDLFile getLastInfoEntity fileName
+      liftIO $ print result
+      case result of
+        Just infoentity -> [|$(liftInfoEntity infoentity)|] -- Lift InfoEntity as expression
+        Nothing -> fail "Failed to generate InfoEntity"
+
+-- | Helper function to lift InfoEntity to Q Exp (used internally by genFloPoCoInfoEntity)
+liftInfoEntity :: InfoEntity -> Q Exp
+liftInfoEntity (InfoEntity n f p ins outs) =
+  [|InfoEntity $(lift n) $(lift f) $(lift p) $(lift ins) $(lift outs)|]

```

Comment:
Why not create an `instance Lift InfoEntity` where you define the `InfoEntity` type?